File: app/src/main/java/com/hkb48/keepdo/settings/DoneIconDialogFragment.java
package com.hkb48.keepdo.settings;

import android.annotation.SuppressLint;
import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;

import androidx.appcompat.app.AlertDialog;
import androidx.preference.PreferenceDialogFragmentCompat;

import com.hkb48.keepdo.R;

public class DoneIconDialogFragment extends PreferenceDialogFragmentCompat {

    public static DoneIconDialogFragment newInstance(String key) {
        DoneIconDialogFragment fragment = new DoneIconDialogFragment();
        Bundle bundle = new Bundle(1);
        bundle.putString(PreferenceDialogFragmentCompat.ARG_KEY, key);
        fragment.setArguments(bundle);
        return fragment;
    }

    @Override
    protected void onPrepareDialogBuilder(AlertDialog.Builder builder) {
        super.onPrepareDialogBuilder(builder);
    }

    @Override
    protected View onCreateDialogView(Context context) {
        LayoutInflater inflater = LayoutInflater.from(getContext());
        @SuppressLint("InflateParams")
        View view = inflater.inflate(R.layout.done_icon_selector, null);

        DoneIconDialogFragment.OnClickListener listener = new DoneIconDialogFragment.OnClickListener();
        view.findViewById(R.id.done_icon_1).setOnClickListener(listener);
        view.findViewById(R.id.done_icon_2).setOnClickListener(listener);
        view.findViewById(R.id.done_icon_3).setOnClickListener(listener);
        view.findViewById(R.id.done_icon_4).setOnClickListener(listener);
        return view;
    }

    @Override
    public void onDialogClosed(boolean positiveResult) {
        // Do nothing
    }

    private class OnClickListener implements View.OnClickListener {
        public void onClick(View v) {
            requireDialog().dismiss();

            String newValue;
            int id = v.getId();
            if (id == R.id.done_icon_2) {
                newValue = "type2";
            } else if (id == R.id.done_icon_3) {
                newValue = "type3";
            } else if (id == R.id.done_icon_4) {
                newValue = "type4";
            } else {
                newValue = "type1";
            }

            DoneIconPreference preference = (DoneIconPreference) getPreference();
            preference.updatePreference(newValue);
        }
    }
}

File: app/src/main/java/com/hkb48/keepdo/settings/DoneIconDialogFragment.kt
package com.hkb48.keepdo.settings

import android.content.Context
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AlertDialog
import androidx.preference.PreferenceDialogFragmentCompat
import com.hkb48.keepdo.R

class DoneIconDialogFragment : PreferenceDialogFragmentCompat() {
    override fun onPrepareDialogBuilder(builder: AlertDialog.Builder) {
        super.onPrepareDialogBuilder(builder)
    }

    override fun onCreateDialogView(context: Context): View {
        val listener = OnClickListener()
        return View.inflate(requireContext(), R.layout.done_icon_selector, null).apply {
            findViewById<View>(R.id.done_icon_1).setOnClickListener(listener)
            findViewById<View>(R.id.done_icon_2).setOnClickListener(listener)
            findViewById<View>(R.id.done_icon_3).setOnClickListener(listener)
            findViewById<View>(R.id.done_icon_4).setOnClickListener(listener)
        }
    }

    override fun onDialogClosed(positiveResult: Boolean) {
        // Do nothing
    }

    private inner class OnClickListener : View.OnClickListener {
        override fun onClick(v: View) {
            requireDialog().dismiss()
            val newValue: String = when (v.id) {
                R.id.done_icon_2 -> "type2"
                R.id.done_icon_3 -> "type3"
                R.id.done_icon_4 -> "type4"
                else -> "type1"
            }
            (preference as DoneIconPreference).updatePreference(newValue)
        }
    }

    companion object {
        fun newInstance(key: String?): DoneIconDialogFragment {
            val fragment = DoneIconDialogFragment()
            val bundle = Bundle(1)
            bundle.putString(ARG_KEY, key)
            fragment.arguments = bundle
            return fragment
        }
    }
}